File: publicq-main/src/PublicQ.Infrastructure/Options/PasswordPolicyOptions.cs
namespace PublicQ.Infrastructure.Options;

/// <summary>
/// Password policy.
/// </summary>
public class PasswordPolicyOptions
{
    /// <summary>
    /// Required length of the password.
    /// </summary>
    public int RequiredLength { get; set; }
    
    /// <summary>
    /// Require digit in the password.
    /// </summary>
    public bool RequireDigit { get; set; }
    
    /// <summary>
    /// Require uppercase letter in the password.
    /// </summary>
    public bool RequireUppercase { get; set; }
    
    /// <summary>
    /// Require lowercase letter in the password.
    /// </summary>
    public bool RequireLowercase { get; set; }
    
    /// <summary>
    /// Require non-alphanumeric character in the password.
    /// </summary>
    public bool RequireNonAlphanumeric { get; set; }
}

File: publicq-main/client/src/models/password-policy-options.ts
/**
 * Password policy configuration options
 */
export interface PasswordPolicyOptions {
  /**
   * Required length of the password
   */
  requiredLength: number;
  
  /**
   * Require digit in the password
   */
  requireDigit: boolean;
  
  /**
   * Require uppercase letter in the password
   */
  requireUppercase: boolean;
  
  /**
   * Require lowercase letter in the password
   */
  requireLowercase: boolean;
  
  /**
   * Require non-alphanumeric character in the password
   */
  requireNonAlphanumeric: boolean;
}